src/domain/wallet-invoices/wallet-invoice-factory.types.d.ts

```diff
@@ -1,8 +1,12 @@
 type WalletInvoiceFactory = {
-  create(registeredInvoice: RegisteredInvoice): WalletInvoice
-  createForRecipient(registeredInvoice: RegisteredInvoice): WalletInvoice
+  createForSelf(
+    registeredInvoice: RegisteredInvoice,
+  ): (fiat: FiatAmount | null) => WalletInvoice
```

Comment:
From scanning the code I have not seen why the currying is necessary. All calls I can find are of the form:
```
  const walletInvoice = walletInvoiceCreateFn(registeredInvoice)(fiat)
```
While I think currying can be very useful I don't understand the need for it here.

Reply:
> From scanning the code I have not seen why the currying is necessary

it's because the `price` is not been fetched initially in the top level use case function (`addInvoiceByWalletId` or `addInvoiceNoAmountByWalletId`). we're doing it an inner function. agree this may be not the most readable code

Reply:
the main things is there are 3 different variables:
- with/without amount
- fiat or btc
- forSelf for forRecipient

we could write the function 8 times but the current code/currying reflect the refactoring of those multi variables inputs.

Reply:
@nicolasburtey will this be updated in regards to what we talk about in daily?

Reply:
yup: 347996c5c74db94e104286538314b716d1106110